msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/JwtHelper.java

```diff
@@ -22,62 +25,183 @@ static ClientAssertion buildJwt(String clientId, final ClientCertificate credent
         ParameterValidationUtils.validateNotNull("credential", clientId);
 
         try {
-            final long time = System.currentTimeMillis();
-
-            // Build header
-            Map<String, Object> header = new HashMap<>();
-            header.put("alg", "RS256");
-            header.put("typ", "JWT");
-
-            if (sendX5c) {
-                List<String> certs = new ArrayList<>(credential.getEncodedPublicKeyCertificateChain());
-                header.put("x5c", certs);
-            }
-
-            //SHA-256 is preferred, however certain flows still require SHA-1 due to what is supported server-side. If SHA-256
-            // is not supported or the IClientCredential.publicCertificateHash256() method is not implemented, the library will default to SHA-1.
-            String hash256 = credential.publicCertificateHash256();
-            if (useSha1 || hash256 == null) {
-                header.put("x5t", credential.publicCertificateHash());
-            } else {
-                header.put("x5t#S256", hash256);
+            // First try with PS256 (preferred)
+            return generatePS256Jwt(clientId, credential, jwtAudience, sendX5c, useSha1);
+        } catch (InvalidKeyException e) {
+            // If the key isn't compatible with PSS, fall back to RS256.
```

Comment:
This is very strange, because the padding algorithm can be always be used with a PSS key

<img width="1069" height="191" alt="image" src="https://github.com/user-attachments/assets/66482e24-9316-4d68-8d5a-e8475206910b" />